homeassistant/components/sunsynkweb/coordinator.py

```diff
@@ -0,0 +1,62 @@
+"""Coordinator for the sunsynk web api."""
+
+from __future__ import annotations
+
+from asyncio import timeout as async_timeout
+from datetime import timedelta
+import logging
+from typing import TYPE_CHECKING
+
+from pysunsynkweb.model import get_plants
+from pysunsynkweb.session import SunsynkwebSession
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+if TYPE_CHECKING:
+    from pysunsynkweb import SunsynkConfigEntry
+
+
+class SunsynkUpdateCoordinator(DataUpdateCoordinator[None]):
+    """A Coordinator that updates sunsynk plants."""
+
+    def __init__(self, hass: HomeAssistant, config: SunsynkConfigEntry) -> None:
```

Comment:
```suggestion
    def __init__(self, hass: HomeAssistant) -> None:
```
ah, now I get the circular import. There is no need to pass the config entry, the coordinator will have that set automatically in a variable `config_entry` after `super().__init__`

Reply:
that makes it a whole lot simpler :) thanks